src/engines/btree.cc

```diff
@@ -49,24 +49,32 @@ using pmem::detail::conditional_add_to_tx;
 namespace pmemkv {
 namespace btree {
 
-BTreeEngine::BTreeEngine(const string& path, const size_t size) {
-    if ((access(path.c_str(), F_OK) != 0) && (size > 0)) {
-        LOG("Creating filesystem pool, path=" << path << ", size=" << to_string(size));
-        pmpool = pool<RootData>::create(path.c_str(), LAYOUT, size, S_IRWXU);
-    } else {
-        LOG("Opening pool, path=" << path);
-        pmpool = pool<RootData>::open(path.c_str(), LAYOUT);
-    }
+BTreeEngine::BTreeEngine(const string& path, const size_t size, const Options options) {
```

Comment:
Using const for non-reference/non-pointer arguments (size & options here) does not make much sense...